test_util/marathon.py

```diff
@@ -205,6 +205,83 @@ def _pool_for_marathon_app(app_id):
             raise Exception("Application deployment failed - operation was not "
                             "completed in {} seconds.".format(timeout))
 
+    def deploy_pod(self, pod_definition):
+        """Deploy a pod to marathon
+
+        This function deploys an a pod and then waits for marathon to
+        aknowledge it's successfull creation or fails the test.
+
+        It waits until all the instances reach tasksRunning and then tasksHealthy state.
+
+        Args:
+            pod_definition: a dict with pod definition as specified in
+                            Marathon API
+            check_health: wait until Marathon reports tasks as healthy before
+                          returning
+        Returns:
+            Scaling instance count
+        """
+        timeout = 120
+
+        r = self.post('v2/pods', json=pod_definition)
+        logging.info('Response from marathon: {}'.format(repr(r.json())))
+        assert r.ok, 'status_code: {} body: {}'.format(r.status_code, r.body)
+
+        @retrying.retry(wait_fixed=2000, stop_max_delay=timeout * 1000,
+                        retry_on_result=lambda ret: ret is False,
+                        retry_on_exception=lambda x: False)
+        def _wait_for_pod_deployment(pod_id):
+            r = self.get('v2/deployments')
+            data = r.json()
+            if len(data) > 0:
+                logging.info('Waiting for pod to be deployed %s', repr(data))
+                return False
+            # deployment complete
+            r = self.get('v2/pods' + pod_id)
+            r.raise_for_status()
+            data = r.json()
+            if int(data['scaling']['instances']) != pod_definition['scaling']['instances']:
+                logging.info('Pod is still scaling. Continuing to wait...')
+                return False
+            return data
+        try:
+            return _wait_for_pod_deployment(pod_definition['id'])
+        except retrying.RetryError as ex:
+            raise Exception("Pod deployment failed - operation was not "
+                            "completed in {} seconds.".format(timeout)) from ex
+
+    def destroy_pod(self, pod_id, timeout=120):
+        """Remove a marathon pod
+
+        Abort the test if the removal was unsuccesful.
+
+        Args:
+            pod_id: id of the pod to remove
+            timeout: seconds to wait for destruction before failing test
+        """
+        @retrying.retry(wait_fixed=1000, stop_max_delay=timeout * 1000,
+                        retry_on_result=lambda ret: not ret,
+                        retry_on_exception=lambda x: False)
+        def _destroy_pod_complete(deployment_id):
+            r = self.get('v2/deployments')
+            assert r.ok, 'status_code: {} body: {}'.format(r.status_code, r.body)
+
+            for deployment in r.json():
```

Comment:
Thanks for filing that @greggomann . If you touch this code again, I would recommend adding the JIRA reference into a code comment just so that its harder to forget about